File: app/lib/shared/data/plans.ts
import { Plan } from "../../domain/Plan";
import { getProblemById } from "./problems";
import { getProfileByName } from "./profiles";

export const INITIAL_PLANS: Array<Plan> = [
  new Plan(1, "Plan 1", getProfileByName("Profile 1"), [
    [true, getProblemById(1)],
    [true, getProblemById(2)],
    [true, getProblemById(3)],
    [true, getProblemById(4)],
    [false, getProblemById(5)],
    [false, getProblemById(8)],
    [false, getProblemById(12)],
    [false, getProblemById(18)],
  ]),
  new Plan(2, "Plan 2", getProfileByName("Profile 2"), [
    [true, getProblemById(11)],
    [true, getProblemById(12)],
    [false, getProblemById(13)],
    [false, getProblemById(14)],
    [false, getProblemById(15)],
    [false, getProblemById(16)],
    [false, getProblemById(17)],
    [false, getProblemById(18)],
  ]),
  new Plan(3, "Plan 3", getProfileByName("Profile 2"), [
    [true, getProblemById(18)],
    [true, getProblemById(19)],
    [false, getProblemById(20)],
    [false, getProblemById(21)],
    [false, getProblemById(22)],
    [false, getProblemById(23)],
    [false, getProblemById(24)],
    [false, getProblemById(25)],
    [false, getProblemById(26)],
    [false, getProblemById(27)],
    [false, getProblemById(28)],
    [false, getProblemById(29)],
  ]),
  new Plan(4, "Plan 4", getProfileByName("Profile 1"), [
    [true, getProblemById(4)],
    [true, getProblemById(5)],
  ]),
  new Plan(5, "Plan 5", getProfileByName("Profile 2"), [
    [true, getProblemById(18)],
    [true, getProblemById(19)],
    [true, getProblemById(20)],
    [false, getProblemById(21)],
    [false, getProblemById(29)],
  ]),
  new Plan(6, "Plan 6", getProfileByName("Profile 1"), [
    [false, getProblemById(26)],
    [false, getProblemById(27)],
    [false, getProblemById(28)],
    [false, getProblemById(29)],
  ]),
  new Plan(7, "Plan 7", getProfileByName("Profile 1"), [
    [true, getProblemById(1)],
    [true, getProblemById(2)],
    [true, getProblemById(8)],
    [true, getProblemById(9)],
  ]),
  new Plan(8, "Plan 8", getProfileByName("Profile 1"), [
    [true, getProblemById(12)],
    [true, getProblemById(25)],
    [true, getProblemById(26)],
    [false, getProblemById(29)],
  ]),
  new Plan(9, "Plan 9", getProfileByName("Profile 2"), [
    [true, getProblemById(12)],
    [true, getProblemById(25)],
    [true, getProblemById(26)],
    [false, getProblemById(29)],
  ]),
  new Plan(10, "Plan 10", getProfileByName("Profile 2"), [
    [true, getProblemById(12)],
    [true, getProblemById(25)],
    [false, getProblemById(26)],
    [false, getProblemById(29)],
  ]),
] as const;

export function getPlansById(id: number): Plan {
  const result = INITIAL_PLANS.find((plan) => plan.id === id);
  if (result === undefined) {
    throw Error();
  }
  return result;
}

File: backend/app/core/data/plans.py
from typing import List
from app.core.data.problems import get_problem_by_id
from app.core.data.profiles import get_profile_by_name
from app.core.domain import Plan

INITIAL_PLANS: List[Plan] = [
    Plan(
        1,
        "Plan 1",
        get_profile_by_name("Profile 1"),
        [
            [True, get_problem_by_id(1)],
            [True, get_problem_by_id(2)],
            [True, get_problem_by_id(3)],
            [True, get_problem_by_id(4)],
            [False, get_problem_by_id(5)],
            [False, get_problem_by_id(8)],
            [False, get_problem_by_id(12)],
            [False, get_problem_by_id(18)],
        ],
    ),
    Plan(
        2,
        "Plan 2",
        get_profile_by_name("Profile 2"),
        [
            [True, get_problem_by_id(11)],
            [True, get_problem_by_id(12)],
            [False, get_problem_by_id(13)],
            [False, get_problem_by_id(14)],
            [False, get_problem_by_id(15)],
            [False, get_problem_by_id(16)],
            [False, get_problem_by_id(17)],
            [False, get_problem_by_id(18)],
        ],
    ),
    Plan(
        3,
        "Plan 3",
        get_profile_by_name("Profile 2"),
        [
            [True, get_problem_by_id(18)],
            [True, get_problem_by_id(19)],
            [False, get_problem_by_id(20)],
            [False, get_problem_by_id(21)],
            [False, get_problem_by_id(22)],
            [False, get_problem_by_id(23)],
            [False, get_problem_by_id(24)],
            [False, get_problem_by_id(25)],
            [False, get_problem_by_id(26)],
            [False, get_problem_by_id(27)],
            [False, get_problem_by_id(28)],
            [False, get_problem_by_id(29)],
        ],
    ),
    Plan(
        4,
        "Plan 4",
        get_profile_by_name("Profile 1"),
        [
            [True, get_problem_by_id(4)],
            [True, get_problem_by_id(5)],
        ],
    ),
    Plan(
        5,
        "Plan 5",
        get_profile_by_name("Profile 2"),
        [
            [True, get_problem_by_id(18)],
            [True, get_problem_by_id(19)],
            [True, get_problem_by_id(20)],
            [False, get_problem_by_id(21)],
            [False, get_problem_by_id(29)],
        ],
    ),
    Plan(
        6,
        "Plan 6",
        get_profile_by_name("Profile 1"),
        [
            [False, get_problem_by_id(26)],
            [False, get_problem_by_id(27)],
            [False, get_problem_by_id(28)],
            [False, get_problem_by_id(29)],
        ],
    ),
    Plan(
        7,
        "Plan 7",
        get_profile_by_name("Profile 1"),
        [
            [True, get_problem_by_id(1)],
            [True, get_problem_by_id(2)],
            [True, get_problem_by_id(8)],
            [True, get_problem_by_id(9)],
        ],
    ),
    Plan(
        8,
        "Plan 8",
        get_profile_by_name("Profile 1"),
        [
            [True, get_problem_by_id(12)],
            [True, get_problem_by_id(25)],
            [True, get_problem_by_id(26)],
            [False, get_problem_by_id(29)],
        ],
    ),
    Plan(
        9,
        "Plan 9",
        get_profile_by_name("Profile 2"),
        [
            [True, get_problem_by_id(12)],
            [True, get_problem_by_id(25)],
            [True, get_problem_by_id(26)],
            [False, get_problem_by_id(29)],
        ],
    ),
    Plan(
        10,
        "Plan 10",
        get_profile_by_name("Profile 2"),
        [
            [True, get_problem_by_id(12)],
            [True, get_problem_by_id(25)],
            [False, get_problem_by_id(26)],
            [False, get_problem_by_id(29)],
        ],
    ),
]


def get_plan_by_id(id: int) -> Plan:
    for plan in INITIAL_PLANS:
        if plan.id == id:
            return plan
    raise ValueError(f"No plan found with id={id}")
